planetiler-core/src/main/java/com/onthegomap/planetiler/util/VarInt.java

```diff
@@ -0,0 +1,129 @@
+
+  private VarInt() {}
+
+  /**
+   * Returns the encoding size in bytes of its input value.
+   *
+   * @param v the long to be measured
+   * @return the encoding size in bytes of a given long value.
+   */
+  public static int varLongSize(long v) {
+    int result = 0;
+    do {
+      result++;
+      v >>>= 7;
+    } while (v != 0);
+    return result;
+  }
+
+  /**
+   * Reads an up to 64 bit long varint from the current position of the given ByteBuffer and returns the decoded value
+   * as long.
+   *
+   * <p>
+   * The position of the buffer is advanced to the first byte after the decoded varint.
+   *
+   * @param src the ByteBuffer to get the var int from
+   * @return The integer value of the decoded long varint
+   */
+  public static long getVarLong(ByteBuffer src) {
+    long tmp;
+    if ((tmp = src.get()) >= 0) {
+      return tmp;
+    }
+    long result = tmp & 0x7f;
+    if ((tmp = src.get()) >= 0) {
+      result |= tmp << 7;
+    } else {
+      result |= (tmp & 0x7f) << 7;
+      if ((tmp = src.get()) >= 0) {
+        result |= tmp << 14;
+      } else {
+        result |= (tmp & 0x7f) << 14;
+        if ((tmp = src.get()) >= 0) {
+          result |= tmp << 21;
+        } else {
+          result |= (tmp & 0x7f) << 21;
+          if ((tmp = src.get()) >= 0) {
+            result |= tmp << 28;
+          } else {
+            result |= (tmp & 0x7f) << 28;
+            if ((tmp = src.get()) >= 0) {
+              result |= tmp << 35;
+            } else {
+              result |= (tmp & 0x7f) << 35;
+              if ((tmp = src.get()) >= 0) {
+                result |= tmp << 42;
+              } else {
+                result |= (tmp & 0x7f) << 42;
+                if ((tmp = src.get()) >= 0) {
+                  result |= tmp << 49;
+                } else {
+                  result |= (tmp & 0x7f) << 49;
+                  if ((tmp = src.get()) >= 0) {
+                    result |= tmp << 56;
+                  } else {
+                    result |= (tmp & 0x7f) << 56;
+                    result |= ((long) src.get()) << 63;
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Encodes a long integer in a variable-length encoding, 7 bits per byte, to a ByteBuffer sink.
+   * 
+   * @param v    the value to encode
+   * @param sink the ByteBuffer to add the encoded value
+   */
+  public static void putVarLong(long v, ByteBuffer sink) {
+    while (true) {
+      int bits = ((int) v) & 0x7f;
+      v >>>= 7;
+      if (v == 0) {
+        sink.put((byte) bits);
+        return;
+      }
+      sink.put((byte) (bits | 0x80));
+    }
+  }
+
+  public static void putVarLong(long v, OutputStream outputStream) throws IOException {
+    byte[] bytes = new byte[varLongSize(v)];
+    ByteBuffer sink = ByteBuffer.wrap(bytes);
+    putVarLong(v, sink);
+    outputStream.write(bytes);
+  }
```

Comment:
It this is going to be the primary method that the pmtiles writer call, I might use the output stream natively to avoid allocating an extra byte array and byte buffer for each insertion. Same as above, except using `outputStream.write(bits | 0x80)` instead of `sink.put((byte) (bits | 0x80))`

Reply:
For reasons unknown the way it's written now is 2x faster (?) on my very synthetic benchmark of 80,000,000 consecutive integers than writing directly to outputStream... not sure why

Reply:
Well I'll eat my words... I  guess you never really know with java until you benchmark it. I think the reason is that `ByteArrayOutputStream.write` (and `BufferedOutputStream.write`) are synchronized, so the fewer times you hit them the better.  To avoid that you'd need to make a smaller byte array, populate it, then write to the output stream but at that point you're basically doing the same thing as making an intermediate byte buffer - so it looks fine as-is!